src/posix/platform/netif.cpp

```diff
@@ -883,33 +896,32 @@ void platformNetifStateChange(otInstance *aInstance, otChangedFlags aFlags)
     }
 }
 
-static void processReceive(otMessage *aMessage, void *aContext)
+static void processReceive(otMessage *aMessage)
 {
-    OT_UNUSED_VARIABLE(aContext);
-
-    char     packet[kMaxIp6Size + 4];
+    char     packetBuffer[kMaxIp6Size + 4 + kIP4IP6HeaderLengthDiff];
```

Comment:
The MTU of this interface is set to `kMaxIp6Size`. Is it possible for the interface to receive a larger IPv4 packet? I think the maximum IPv4 datagram would be (kMaxIp6Size - kIP4IP6HeaderLengthDiff).

Reply:
Yes, this change should be removed. I don't think we need to care about this for now since:
- It won't cause issue when fragmentation is enabled
- OT will drop it if it hits the limit and fragmentation is not enabled.